plugins/jenkins-backend/config.d.ts

```diff
@@ -40,9 +41,16 @@ export interface Config {
       name: string;
 
       baseUrl: string;
+      /** @visibility secret */
       username: string;
       /** @visibility secret */
       apiKey: string;
+      extraRequestHeaders?: {
```

Comment:
Hm this declares that the request headers can only ever contain authentication, and nothing else. Maybe there's no harm in declaring the entire `extraRequestHeaders` secret instead, and setting its type to `Record<string, string>`

Reply:
I think we can't do this as `Record<string, string>` is a object and the @visibility secret has no effect on it.
Updated the version with a union type.

This would work with a configuration like 
```
jenkins:
  instances:
    - name: default
      baseUrl: https://jenkins.example.com
      username: backstage-bot
      apiKey: 123456789abcdef0123456789abcedf012
    - name: bubu
      baseUrl: https://jenkins.example.com
      username: backstage-bot
      apiKey: 123456789abcdef0123456789abcedf012
      extraRequestHeaders:
        Authentication: bubu
        lala: same-same
```